crates/iceberg/src/spec/manifest.rs

```diff
@@ -0,0 +1,1864 @@
+                        Type::Primitive(PrimitiveType::Long),
+                    )),
+                }),
+            ))
+        })
+    };
+
+    static VALUE_COUNTS: Lazy<NestedFieldRef> = {
+        Lazy::new(|| {
+            Arc::new(NestedField::optional(
+                109,
+                "value_counts",
+                Type::Map(MapType {
+                    key_field: Arc::new(NestedField::required(
+                        119,
+                        "key",
+                        Type::Primitive(PrimitiveType::Int),
+                    )),
+                    value_field: Arc::new(NestedField::required(
+                        120,
+                        "value",
+                        Type::Primitive(PrimitiveType::Long),
+                    )),
+                }),
+            ))
+        })
+    };
+
+    static NULL_VALUE_COUNTS: Lazy<NestedFieldRef> = {
+        Lazy::new(|| {
+            Arc::new(NestedField::optional(
+                110,
+                "null_value_counts",
+                Type::Map(MapType {
+                    key_field: Arc::new(NestedField::required(
+                        121,
+                        "key",
+                        Type::Primitive(PrimitiveType::Int),
+                    )),
+                    value_field: Arc::new(NestedField::required(
+                        122,
+                        "value",
+                        Type::Primitive(PrimitiveType::Long),
+                    )),
+                }),
+            ))
+        })
+    };
+
+    static NAN_VALUE_COUNTS: Lazy<NestedFieldRef> = {
+        Lazy::new(|| {
+            Arc::new(NestedField::optional(
+                137,
+                "nan_value_counts",
+                Type::Map(MapType {
+                    key_field: Arc::new(NestedField::required(
+                        138,
+                        "key",
+                        Type::Primitive(PrimitiveType::Int),
+                    )),
+                    value_field: Arc::new(NestedField::required(
+                        139,
+                        "value",
+                        Type::Primitive(PrimitiveType::Long),
+                    )),
+                }),
+            ))
+        })
+    };
+
+    static DISTINCT_COUNTS: Lazy<NestedFieldRef> = {
+        Lazy::new(|| {
+            Arc::new(NestedField::optional(
+                111,
+                "distinct_counts",
+                Type::Map(MapType {
+                    key_field: Arc::new(NestedField::required(
+                        123,
+                        "key",
+                        Type::Primitive(PrimitiveType::Int),
+                    )),
+                    value_field: Arc::new(NestedField::required(
+                        124,
+                        "value",
+                        Type::Primitive(PrimitiveType::Long),
+                    )),
+                }),
+            ))
+        })
+    };
```

Comment:
Both Java and PyIceberg don't write `distinct_counts`

Reply:
Seems the spec doesn't have info about this. Is that a consensus to deprecatei?🤔

Reply:
Yes, it was never implemented in most engines since it is not cheap to compute this kind of statistics. The alternative now is to compute sketches and store them in Puffin files.

Reply:
Got it! Thanks!